From fc1e1f4c2212b478b2617d30bcfc89836a47971e Mon Sep 17 00:00:00 2001
From: Kasin Li <quic_donglil@quicinc.com>
Date: Wed, 15 Jun 2016 23:49:31 +0800
Subject: [PATCH 106/107] drm: adreno: Fix issue of zero addrass writing

This issu is caused by CP trying to update RPTR to CP_RPTR_ADDR, which
is null. We don't need this letency function.

Change-Id: I416da730c97815bfd318581d6b3a4f412c28839b
Signed-off-by: Kasin Li <quic_donglil@quicinc.com>
---
 drivers/gpu/drm/msm/adreno/adreno_gpu.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/msm/adreno/adreno_gpu.c b/drivers/gpu/drm/msm/adreno/adreno_gpu.c
index 4f9cfb9..5d63a0a 100644
--- a/drivers/gpu/drm/msm/adreno/adreno_gpu.c
+++ b/drivers/gpu/drm/msm/adreno/adreno_gpu.c
@@ -68,7 +68,8 @@ int adreno_hw_init(struct msm_gpu *gpu)
 	/* Setup REG_CP_RB_CNTL: */
 	adreno_gpu_write(adreno_gpu, REG_ADRENO_CP_RB_CNTL,
 			/* size is log2(quad-words): */
-			AXXX_CP_RB_CNTL_BUFSZ(ilog2(gpu->rb->size / 8)));
+			AXXX_CP_RB_CNTL_BUFSZ(ilog2(gpu->rb->size / 8))|
+			(1 << 27));
 
 	/* Setup ringbuffer address: */
 	adreno_gpu_write(adreno_gpu, REG_ADRENO_CP_RB_BASE,
-- 
2.8.2

